File: src/test/java/org/cerion/stocklist/indicators/DirectionalIndexTest.java
package org.cerion.stocklist.indicators;

import org.cerion.stocklist.arrays.PairArray;
import org.cerion.stocklist.functions.FunctionTestBase;
import org.junit.Test;

public class DirectionalIndexTest extends FunctionTestBase {

    @Test
    public void directionalIndex_defaults() {
        PairArray arr = new DirectionalIndex().eval(mPriceList);

        int last = arr.size();

        // TODO verify values online, just doing these pre-refactor
        assertEqual(30.29, arr.getNeg(last-1), "last");
        assertEqual(27.01, arr.getPos(last-1), "last");
    }
}

File: src/test/kotlin/indicators/DirectionalIndexTest.kt
package org.cerion.stocks.core.indicators

import org.cerion.stocks.core.TestBase
import org.junit.Test

class DirectionalIndexTest : TestBase() {

    @Test
    fun directionalIndex_defaults() {
        val arr = DirectionalIndex().eval(priceList)
        val last = arr.size - 1

        // TODO verify values online, just doing these pre-refactor
        assertEqual(30.29, arr.neg(last), "last")
        assertEqual(27.01, arr.pos(last), "last")
    }
}